src/libraries/System.Text.Json/System.Text.Json.sln

```diff
@@ -35,6 +35,20 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ref", "ref", "{EC8CE194-261
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{676B6044-FA47-4B7D-AEC2-FA94DB23A423}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "gen", "gen", "{A403699B-1BB2-4B23-AE05-97608C23EB27}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{0DBD3DDF-16F1-4E7D-80A5-F9D74C1219B7}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Text.Json.SourceGeneration", "gen\src\System.Text.Json.SourceGeneration.csproj", "{24D86F06-B8C4-4A26-A6D4-1997FA0DD2FF}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{C4C10F31-F6DF-40DB-B892-CDFDD4DD9091}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Text.Json.SourceGeneration.Tests", "gen\tests\System.Text.Json.SourceGeneration.Tests.csproj", "{9DE9B0B4-8721-4A6A-B73C-59D89E5212B9}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "unit_tests", "unit_tests", "{7B11F28A-2732-49CF-8000-682FBDE606A7}"
```

Comment:
I don't think we follow this naming convention or structure for projects containing multiple test projects. See  for example:

https://github.com/dotnet/runtime/tree/main/src/libraries/System.Console/tests/ManualTests
https://github.com/dotnet/runtime/tree/main/src/libraries/System.Formats.Cbor/tests/CborDocument